### ISSUE TITLE
Notification close button is missing a label for screen readers

### ISSUE THREAD

**Issue by Mel-Roth-AI** [telerik/kendo-angular#3206]
**Describe the bug**
The Notification close button has a title, but not all screen readers read the title. 

**To Reproduce**
1. Visit https://www.telerik.com/kendo-angular-ui/components/notification/
2. Click the Save data button to trigger a notification
3. View source on the close x icon.

Source: <a class="k-icon k-i-close" title="Close"></a>
The anchor only has a title. No hidden label, and no ARIA label.

**Expected behavior**
Option 1:
Include aria-label:
<a class="k-icon k-i-close" aria-label="Close" title="Close"></a>

Option 2:
Include a label that is hidden using CSS:
<a class="k-icon k-i-close"><span class="screen-reader-only">Close</span></a>

**Additional context**
Reference for which screen readers read title attribute https://www.powermapper.com/tests/screen-readers/labelling/a-title/

**Comment by slvy** [comment]
Identical bug/requested fix for k-i-filter
